sgl-router/src/tool_parser/parsers/qwen_coder.rs

```diff
@@ -0,0 +1,598 @@
+                                Value::String(value.to_string())
+                            }
+                        };
+
+                        // Add to current parameters
+                        self.xml_current_parameters
+                            .insert(key.clone(), json_value.clone());
+
+                        // Stream the parameter update
+                        let value_json = serde_json::to_string(&json_value)
+                            .map_err(|e| ParserError::ParsingFailed(e.to_string()))?;
+
+                        let json_fragment = if self.xml_streamed_parameters.is_empty() {
+                            format!("{{\"{}\": {}}}", key, value_json)
+                        } else {
+                            format!(", \"{}\": {}", key, value_json)
+                        };
+
+                        calls.push(ToolCallItem {
+                            tool_index: self.current_tool_id as usize,
+                            name: None,
+                            parameters: json_fragment.clone(),
+                        });
+
+                        // Update streamed args
+                        let current_args =
+                            &mut self.streamed_args_for_tool[self.current_tool_id as usize];
+                        if current_args.is_empty() {
+                            *current_args = format!("{{\"{}\": {}}}", key, value_json);
+                        } else {
+                            // Trim trailing whitespace before checking for closing brace
+                            let trimmed = current_args.trim_end();
+                            if let Some(stripped) = trimmed.strip_suffix('}') {
+                                *current_args = format!("{}{}}}", stripped, json_fragment);
+                            } else {
+                                *current_args = format!("{}{}", trimmed, json_fragment);
+                            }
+                        }
+
+                        // Update streamed parameters
+                        self.xml_streamed_parameters.insert(key, json_value);
+                    }
+                }
+            }
+
+            // Use precompiled regex pattern
+            // Check if we're entering a new parameter
+            if !self.in_parameter {
+                if let Some(cap) = self.xml_param_start_pattern.captures(&self.buffer) {
+                    if let Some(key_match) = cap.get(1) {
+                        self.current_parameter_key = key_match.as_str().trim().to_string();
+                        self.current_parameter_value.clear();
+                        self.in_parameter = true;
+
+                        // Remove the opening tag from buffer
+                        if let Some(m) = cap.get(0) {
+                            self.buffer = self.buffer[m.end()..].to_string();
+                        }
+                    }
+                }
+            }
```

Comment:
Why do aggresively remove the open tag `<parameter` here? We don't need L333 to L437, no?

The accumulation there is no different than the fast path on L257 will be able to capture this eventually.

Unless we are streaming parameters incrementally, I don't see why we need this.